### PR TITLE
aggregation availability should ensure that discovery responds non-failing

### PR DESCRIPTION
**Pull request by deads2k**
An apiservice is checked for healthy discovery before being marked available.  Previously, we forgot to the status code of the discovery before marking it.  This updates us to ensure that the apiservice will respond with a non-failing return code.

/kind bug
@kubernetes/sig-api-machinery-bugs 
/priority important-soon

```release-note
NONE
```

### CODE REVIEW

**Review by logicalhan — COMMENTED** [review]
I think this generally makes sense. Would it be worthwhile to add a testcase for this?

**Comment by deads2k** [comment]
godep again

/retest

**Comment by sttts** [comment]
Forgotten to add the test? I don't see it.

File: staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller_test.go [hunk 1]
```diff
@@ -18,11 +18,15 @@ package apiserver
...
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
 	"testing"
 
 	"github.com/davecgh/go-spew/spew"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (sttts) [inline staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller_test.go]: nit: not necessary

**Comment by deads2k** [comment]
hrm.  If these fail, need to hold to figure out what changed about that test.  Worked before I added my unit test :)

**Comment by deads2k** [comment]
must have been flaky.

File: staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go [hunk 2]
```diff
@@ -254,17 +264,31 @@ func (c *AvailableConditionController) sync(key string) error {
...
+						return
+					}
+
+					// setting the system-masters identity ensures that we will always have access rights
+					transport.SetAuthProxyHeaders(newReq, "system:kube-aggregator", []string{"system:masters"}, nil)
+					resp, err := c.discoveryClient.Do(newReq)
 					if resp != nil {
 						resp.Body.Close()
+						// we should always been in the 200s or 300s
+						if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
```
Comment (tedyu) [inline staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go]: Looking at http.go, it seems code 300 is not treated as error:
```
		if res.StatusCode >= http.StatusMultipleChoices { // Redirect
			klog.V(4).Infof("Probe terminated redirects for %s, Response: %v", url.String(), *res)
			return probe.Warning, body, nil
```

**Pull request merged**